src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,489 @@
+                this.textWriter.Dispose();
+                this.Flush();
+
+                CheckIfSeparatorNeeded();
+                CheckIfManualValueAtArrayStart();
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously starts a TextWriter value scope with the specified content type.
+        /// </summary>
+        /// <param name="contentType">The content type of the TextWriter value scope.</param>
+        /// <returns>A task representing the asynchronous operation. The task result is a TextWriter for writing JSON data.</returns>
+        public async Task<TextWriter> StartTextWriterValueScopeAsync(string contentType)
+        {
+            await this.FlushAsync().ConfigureAwait(false);
+
+            WriteItemWithSeparatorIfNeeded();
+
+            this.currentTextWriterContentType = contentType;
+
+            bool isWritingJson = String.IsNullOrEmpty(this.currentTextWriterContentType) || this.currentTextWriterContentType.StartsWith(MimeConstants.MimeApplicationJson, StringComparison.Ordinal);
+
+            if (!isWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            }
+
+            await this.FlushAsync().ConfigureAwait(false);
+
+            this.textWriter = new ODataUtf8JsonTextWriter(this, isWritingJson);
+
+            return this.textWriter;
+        }
+
+        /// <summary>
+        /// Asynchronously ends a TextWriter value scope.
+        /// </summary>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public async Task EndTextWriterValueScopeAsync()
+        {
+            bool isWritingJson = String.IsNullOrEmpty(this.currentTextWriterContentType) || this.currentTextWriterContentType.StartsWith(MimeConstants.MimeApplicationJson, StringComparison.Ordinal);
+
+            if (!isWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+                await this.textWriter.DisposeAsync();
+                await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+
+                CheckIfSeparatorNeeded();
+                CheckIfManualValueAtArrayStart();
+            }
+        }
+
+        /// <summary>
+        /// Represents a TextWriter implementation for writing UTF-8 JSON data in an OData context.
+        /// </summary>
+        private sealed class ODataUtf8JsonTextWriter : TextWriter
+        {
+            ODataUtf8JsonWriter jsonWriter = null;
+            private char[] buffer;
+            private bool isWritingJson;
+            private int numOfCharsNotWrittenFromPreviousChunk = 0;
+
+            public ODataUtf8JsonTextWriter(ODataUtf8JsonWriter jsonWriter, bool isWritingJson)
+            {
+                this.jsonWriter = jsonWriter;
+                this.isWritingJson = isWritingJson;
+            }
+
+            public override Encoding Encoding => throw new NotImplementedException();

```

Comment:
```suggestion
            public override Encoding Encoding => Encoding.UTF8;
```
Maybe?

Or if not,
```csharp
        <summary>
        Gets the character encoding in which the output is written. This is not supported by this text writer.
        </summary>
        public override Encoding Encoding => throw NotSupportedException();
```